activesupport/lib/active_support/inflector/transliterate.rb

```diff
@@ -66,7 +66,7 @@ def transliterate(string, replacement = "?", locale: nil)
       raise ArgumentError, "Cannot transliterate strings with #{string.encoding} encoding" unless ALLOWED_ENCODINGS_FOR_TRANSLITERATE.include?(string.encoding)
 
       string = string.dup if string.frozen?
-      return string if string.ascii_only?
+      return string.dup if string.ascii_only?
```

Comment:
Now there's a double `dup` in case the string is frozen **and** ascii_only. The return could be moved before the "frozen dup" to prevent that.

Reply:
Good call! Updated in df7df11ad2665ea5b3a279c7257ab856ca6220eb.